src/Coordination/KeeperServer.cpp

```diff
@@ -921,6 +924,9 @@ nuraft::cb_func::ReturnCode KeeperServer::callbackFunc(nuraft::cb_func::Type typ
                 if (request_for_session->digest->version != KeeperStorageBase::NO_DIGEST)
                     writeIntBinary(request_for_session->digest->value, write_buf);
 
+                if (serialization_version < IKeeperStateMachine::ZooKeeperLogSerializationVersion::WITH_XID_64)
+                    writeIntBinary(static_cast<uint32_t>(0), write_buf);
+
```

Comment:
Maybe because we always want to write 64bit IDs to have homogeneous log entries? In that case, if we have to revert to older version of keeper, can the older versions read the new format (assuming only 32bit IDs were used)?

Reply:
After discussing this on slack I got to know that before we left this word uninitialized, now it is filled with zero.